### PR TITLE
Supplemental Metrics for ECG Reading

### PR DESCRIPTION
**Pull request by MatthewTurk247**
# Supplemental Metrics for ECG Reading

## :recycle: Current situation & Problem
Building on top of #40, this pull request seeks to package ECG readings with supplemental information like VO2 max and pulse rate in a way that is sensitive to the possible limitations of current recording capabilities.

## :gear: Release Notes 
- Measures most recent VO2 max known at the time of ECG recording.
- Captures pulse rate via optical sensor at T-minus five minutes and T-plus five minutes.
- Captures activity metrics at T-minus five minutes and T-plus five minutes, inlcuding:
  - Active calories.
  - Step count.
  - Standing minutes.
- Collects readings of heart rate from optical sensor at intervals of 10 minutes for first two days of the study.
- Estimates amount of time watch has been worn each day.

## :books: Documentation
In-line documentation will be written in relevant files in conformance to the [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).

## :white_check_mark: Testing
Tests may be added for ensuring that data is correctly serialized and properly handled.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MatthewTurk247** [comment]
Some questions for @PSchmiedmayer:
- Should I create a new `add` function in `PAWSStandard` or modify the existing one that handles `HKElectrocardiogram`?
- As I understand it, we want to capture the most recent VO2 max at the time of ECG recording. Should that code go in one of the existing files in `PAWS/ECGRecordings`?
- If so, do we need a new data structure for associating ECG readings with VO2 samples? I suppose that without relating the two, we could still query in such a way as to construct pairings of data if we have the timestamps, but I am curious to hear additional perspectives on this matter.

**Comment by PSchmiedmayer** [comment]
@MatthewTurk247 Sorry for the late response! 

While SpeziHealthKit is great to collect longitudinal data, I would suggest to use normal "custom" HealthKit queries for the 5 minute intervals around the measurements.
You can use the Spezi Delegate HealthKit configuration with a `.manual()` delivery setting to make sure that permissions are granted during the onboarding but I would suggest that we develop a custom HealthKit query that gets triggered when a new ECG is recorded and received in the Standard that would then trigger the surrounding data collection.

I think it is fine to just store them in the database as they can be correlated with the ECG using the timestamps. In the long run we could work with FHIR references between the different documents but that might be too complicated for now. Just storing them as observations should be fine and should address our need.

We would also need some additional logic around the correlation of incoming symptoms and how to correlate this with the ECG types.

Let's use the meeting today to take a deep-dive into some of the implementation ideas and how we can address them as part of this and future PRs.

File: PAWS/ECGRecordings/ECGRecording.swift
Comment (PSchmiedmayer) [inline PAWS/ECGRecordings/ECGRecording.swift:1]: Very good use of the async HealthKit functions and great to see the reusability in the computed properties, great job!!

File: PAWS/PAWSDelegate.swift [hunk 1]
```diff
@@ -77,7 +77,7 @@ class PAWSDelegate: SpeziAppDelegate {
         // Collection starts at the time the user consents and lasts for 1 month.
         let sharedPredicate = HKQuery.predicateForSamples(
             withStart: healthKitStartDate,
-            end: Calendar.current.date(byAdding: DateComponents(month: 1), to: healthKitStartDate),
+            end: Calendar.current.date(byAdding: DateComponents(month: 6), to: healthKitStartDate),
```
Comment (PSchmiedmayer) [inline PAWS/PAWSDelegate.swift:80]: Nice; thank you for fixing this here 🚀

File: PAWS/PAWSStandard.swift
Comment (PSchmiedmayer) [inline PAWS/PAWSStandard.swift:1]: Very cool job; looks great!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements @MatthewTurk247; the changes look great and are well structured! 🚀

Great job with your first PR!

**Comment by PSchmiedmayer** [comment]
@MatthewTurk247 I would merge the PR if you agree that you don't want to make any additional changes? I can override the missing patch requirement. It will be nearly impossible to easily unit test them with the current testing capabilities that are available for HealthKit.

**Comment by MatthewTurk247** [comment]
> @MatthewTurk247 I would merge the PR if you agree that you don't want to make any additional changes? I can override the missing patch requirement. It will be nearly impossible to easily unit test them with the current testing capabilities that are available for HealthKit.

Yep, sounds good to me! I don't have any other changes in mind for this specific PR. So once that missing patch requirement is overridden and the PR is merged, I think it will make sense to open a new PR specifically for symptom tagging and linking to ECG recordings.

**Comment by PSchmiedmayer** [comment]
Sounds great; I have merged the PR 🎉

**Comment by PSchmiedmayer** [comment]
Thank you for all the work and research that went into the PR @MatthewTurk247 🚀

**Pull request merged**